src/main/java/trade/wayruha/mexc/ws/WebSocketClient.java

```diff
@@ -72,77 +76,103 @@ public class WebSocketClient<T> extends WebSocketListener {
+            if (this.webSocket == null) {
+                log.error("Error connect {}", this.connectionRequest);
+            }
         } else {
-            log.warn("{} Already connected", logPrefix);
+            log.warn("{} Already connected to channels {}", logPrefix, channels);
         }
         if (!newChannels.isEmpty()) {
             this.subscribe(newChannels);
-            this.scheduledPingTask = this.scheduler.scheduleAtFixedRate(new PingTask(newChannels),
-                    WEB_SOCKET_KEEP_ALIVE_TOPIC_PERIOD_SEC, WEB_SOCKET_KEEP_ALIVE_TOPIC_PERIOD_SEC, TimeUnit.SECONDS);
+            //create ping task for keep alive WS connection if no messages were produced be server
+            this.scheduledPingTask = scheduler.scheduleAtFixedRate(new PingTask(),
+                    this.config.getWebSocketChannelKeepAlivePeriodSec(), this.config.getWebSocketChannelKeepAlivePeriodSec(), TimeUnit.SECONDS);
         }
+        currentReconnectAttemptIndex.set(0);//reset reconnection indexer due to successful connection
     }
 
-    public void send(String str) {
+    public void sendRequest(String request) {
         boolean result = false;
-        log.debug("{}:send() {}", logPrefix, str);
+        log.debug("{} Try to send request {}", logPrefix, request);
         if (webSocket != null) {
-            result = webSocket.send(str);
+            result = webSocket.send(request);
         }
         if (!result) {
-            log.error("{}: Failed to send message: {}", logPrefix, str);
+            log.error("{} Failed to send message: {}", logPrefix, request);
             closeOnError(null);
         }
     }
 
     @SneakyThrows
     public void subscribe(Collection<String> newChannels) {
-        if (this.channels.size() + newChannels.size() > WEB_SOCKET_MAX_CHANNELS_PER_CONNECTION)
-            throw new IllegalArgumentException("Can't subscribe to more then " + WEB_SOCKET_MAX_CHANNELS_PER_CONNECTION + "channels in single WS connection");
+        if (this.channels.size() + newChannels.size() > this.config.getWebSocketMaxReconnectAttemptNumber())
+            throw new IllegalArgumentException("Can't subscribe to more then " + WEB_SOCKET_MAX_CHANNELS_PER_CONNECTION + " channels in single WS connection");
         final Message msg = new Message(Action.SUBSCRIPTION, newChannels);
         final String content = objectMapper.writeValueAsString(msg);
-        this.send(content);
+        this.sendRequest(content);
         this.channels.addAll(newChannels);
     }
 
     @SneakyThrows
     public void ping(Collection<String> channels) {
         final Message msg = new Message(Action.PING, channels);
         final String content = objectMapper.writeValueAsString(msg);
-        this.send(content);
-        log.debug("{} ping was send to channels {} ", logPrefix, channels);
+        this.sendRequest(content);
+        log.trace("{} Ping was send to channels {} ", logPrefix, channels);
```

Comment:
this log may be redundant as we also do log every request in sendRequest().

Reply:
Here I used trace level for logging. Let it be here for detailed debugging.